### PR TITLE
Dispatch jobs without relating to models

### PR DESCRIPTION
**Pull request by mateusjunges**
This PR changes how relating a tracked job to a model works. 

In v1, you must add a `Illuminate\Database\Eloquent\Model` instance as the first parameter of your's job constructor, which will be used to track the job and relate it to the given model. In this PR, I changed this behavior to have no related models by default. 

You can define the new `trackableKey` and `trackableType` methods in your job class if you need to relate your job to any given model.

For example, if you define these two methods:

```php
public function trackableKey(): ?string
{
    return (string) $this->podcast->id;
}

public function trackableType(): ?string
{
    return $this->podcast->gerMorphClass();
}
```

the `TrackedJob` instance will be created using the podcast id as `trackable_id` and the podcast model FQCN (or any relation morph map you have defined) as the `trackable_type`.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -14,7 +14,8 @@
         "php": "^7.4||^8.0",
```
Comment (mateusjunges) [inline composer.json:14]: I think we can drop support for php 7.4

**Comment by mateusjunges** [comment]
Hey @DimaVIII 👋 I'm quite out of time to work on it right now, that's why it is paused. I plan to ship it on v2, as soon as I have the time to go back to it 🙂

**Comment by DimaVIII** [comment]
Hey @mateusjunges ! 

Based on this PR I added all the mentioned features and fixed all bug, I also added tests for all new features that I implemented. 
https://github.com/DimaVIII/trackable-jobs-for-laravel/tree/features-v2

I'm still testing and a bit more clean up is missing, so haven't pushed the the PR yet.

#### Things to consider
- This Commit https://github.com/mateusjunges/trackable-jobs-for-laravel/commit/31cbbb080154ecdce33a09752dbc0e0be077fce2 is kind of controversial. In my opinion it's not a good idea to publish the migration files, it should stay in the package. I would suggest to revert this in v2. I added a possible solution for this which hopefully should avoid breaking changes.

A small preview on how it looks like implanted into a frontend.
Dashboard

**Comment by bokanechase-digistorm** [comment]
@mateusjunges is it possible to get a new release that includes these changes? 🙏 
The latest release v1.7.0 [doesn't seem to have them](https://github.com/mateusjunges/trackable-jobs-for-laravel/blob/v1.7.0/src/Concerns/Trackable.php).

**Pull request merged**